api/app/signals/apps/signals/models/routing_expression.py

```diff
@@ -14,7 +18,22 @@ class RoutingExpression(models.Model):
         related_name='routing_department'
     )
     _department = models.ForeignKey(Department, on_delete=models.CASCADE)
+    _user = models.ForeignKey(to=User, on_delete=models.SET_NULL, blank=True, null=True)
     order = models.PositiveIntegerField(default=0, db_index=True)
     is_active = models.BooleanField(default=False)
 
-    objects = models.Manager()
+    def clean(self):
+        super(RoutingExpression, self).clean()
+
+        errors = {}
+        if self.is_active and self._user and not self._user.profile.departments.filter(id=self._department.id).exists():
```

Comment:
Do we need to check for inactive users as well?

Reply:
That is also a good one to add. Will do that!